securedrop_client/sync.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+from PyQt5.QtCore import pyqtSignal, QObject, QThread, QTimer, Qt
+from sqlalchemy.orm import scoped_session
+from sdclientapi import API, RequestTimeoutError
+
+from securedrop_client.api_jobs.sync import MetadataSyncJob
+from securedrop_client.crypto import GpgHelper
+
+
+logger = logging.getLogger(__name__)
+
+
+class ApiSync(QObject):
+    '''
+    ApiSync continuously executes a MetadataSyncJob, waiting 15 seconds between jobs.
+    '''
+    sync_started = pyqtSignal()
+    sync_success = pyqtSignal()
+    sync_failure = pyqtSignal(Exception)
+
+    TIME_BETWEEN_SYNCS_MS = 1000 * 15  # fifteen seconds between syncs
+
+    def __init__(
+        self, api_client: API, session_maker: scoped_session, gpg: GpgHelper, data_dir: str
+    ):
+        super().__init__()
+
+        self.api_client = api_client
+        self.session_maker = session_maker
+        self.gpg = gpg
+        self.data_dir = data_dir
+
+        self.sync_thread = QThread()
+        self.sync_thread.started.connect(self._sync)
+
+    def start(self, api_client: API) -> None:
+        '''
+        Start metadata syncs.
+        '''
+        self.api_client = api_client
+
+        if not self.sync_thread.isRunning():
+            logger.debug('Starting sync thread')
+            self.sync_thread.start()
+
+    def stop(self):
+        '''
+        Stop metadata syncs.
+        '''
+        self.api_client = None
+
+        if self.sync_thread.isRunning():
+            logger.debug('Stopping sync thread')
+            self.sync_thread.quit()
+
+    def _sync(self):
+        '''
+        Create and run a new MetadataSyncJob.
+        '''
+        job = MetadataSyncJob(self.data_dir, self.gpg)
+        job.success_signal.connect(self._on_sync_success, type=Qt.QueuedConnection)
+        job.failure_signal.connect(self._on_sync_failure, type=Qt.QueuedConnection)
+
+        session = self.session_maker()
+        job._do_call_api(self.api_client, session)
```

Comment:
I think we'll need some exception handling (similar to https://github.com/freedomofpress/securedrop-client/blob/master/securedrop_client/queue.py#L135) around this since otherwise `_do_call_api` will raise exceptions that will crash the app (i.e. if we logout)